rhp/v3/transport.go

```diff
@@ -95,7 +95,16 @@ func (s *Stream) writeObject(resp *rpcResponse) error {
 	resp.EncodeTo(e)
 	e.Flush()
 	b := buf.Bytes()
-	binary.LittleEndian.PutUint64(b[:8], uint64(len(b)-8))
+	length := len(b) - 8
+	// HACK: in siad, the ExecuteProgram RPC sends the ProgramData separately
+	// from the rest of the request object. This was a performance optimization
+	// relating to how siad handles encoding and length prefixes. Suffice it to
+	// say, that optimization is not necessary in core, but we still have to
+	// preserve compatibility with how siad does things.
+	if r, ok := resp.data.(*RPCExecuteProgramRequest); ok {
```

Comment:
Should this be handled by exposing `Stream.Write` and removing the `Output` field? This feels really hacky

Reply:
That was also what I had in mind but I talked it over with Luke and he really does not want to expose `Stream.Write`, which I definitely understand. It's still writing an RPC object 🤷‍♂️ . I'm usually not really in favour of hacks like this but in this case I don't really mind, it's pretty clear cut.